code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -196,6 +195,15 @@
 			to_chat(src, "<span class='warning'>Something in this place is interfering with our link to Queen Mother. We are unable to evolve to a psychic caste here!</span>")
 			return
 
+	else if(new_caste_type == /mob/living/carbon/xenomorph/hivemind) //Special case for dealing with hiveminds - this may be subject to heavy change, such as multiple hiveminds potentially being an option
+		if(length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/hivemind]))
+			to_chat(src, "<span class='warning'>There cannot be two manifestations of the hivemind's will at once..</span>")
+			return
+
+		if(isxenoresearcharea(get_area(src)))
+			to_chat(src, "<span class='warning'>Something in this place is interfering with our link to the Hivemind. We are unable to evolve to be its manifestation!</span>")
+			return
+
```

Comment:
since a hivemind is like a structure, you should do the same structure checks

Reply:
I dont see the point of doing any structure checks on hiveminds due to the fact you need to A) be a drone before evolving and B) evolving in a place where the structure check for nodes fails will actually be to your detriment, since you couldnt build defences around your core.

Reply:
There are a number of things that would be beneficial to a hivemind but would fail, for example building a hivemind on top a carrier trap or even a resin bed.

Reply:
or... just placed in fog where they are immortal

Reply:
Point is, theres 0 concievable advantage of placing hivemind on normally unplacable tiles, since you could just do the same thing in fog (which you currently can).

Reply:
Thats besides the point, structure creation for aliens currently run the check_alien_construction.

Whether or not it makes sense for the user, for the codebase it allows us to control structure creation from a single place.

I agree there are a lot of things that can be improved regarding the hivemind and im not opposed to them.

This however is seperate, and shouldnt be seen as the same.